src/aig-common/modules/standard/components/ipp-modality/ipp-modality-form/ipp-modality-form.component.ts

```diff
@@ -0,0 +1,83 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { MatSnackBar } from '@angular/material/snack-bar';
+import { FuseProgressBarService } from '@fuse/components/progress-bar/progress-bar.service';
+import { ItalianPublicProcurementModalityResourceService, ItalianPublicProcurementModalityDTO  } from 'aig-standard';
+import { EventService } from 'aig-common/event-manager/event.service';
+
+@Component({
+    selector: 'aig-ipp-modality-form',
+    templateUrl: './ipp-modality-form.component.html',
+    styleUrls: ['./ipp-modality-form.component.scss']
+})
+export class AigIppModalityFormComponent implements OnInit {
+
+    step: any = {
+        form: true,
+        loading: false,
+        complete: false
+    };
+
+    constructor(
+        private _formBuilder: FormBuilder,
+        private _fuseProgressBarService: FuseProgressBarService,
+        private _snackBar: MatSnackBar,
+        private ippModalityResourceService: ItalianPublicProcurementModalityResourceService,
+        private eventService: EventService,
+    ) { }
+    
+    @Input()
+    ippModality: ItalianPublicProcurementModalityDTO;
+
+    ippModalityNewUpdateForm: FormGroup;
+
+    ngOnInit(): void {
+        this.ippModalityNewUpdateForm = this._formBuilder.group({
+            id: [''],
+            name: ['', Validators.required],
+            code: ['', Validators.required],
+            wikiCode:['']
+        })
+
+        if (this.ippModality != null) {
+            this.ippModalityNewUpdateForm.patchValue(this.ippModality);
+        }
+    }
+
+    async submit() {
+        if (!this.ippModalityNewUpdateForm.valid) {
+            return;
+        }
+        this._fuseProgressBarService.show();
+        this.setStep("loading");
+
+        let ippModality: ItalianPublicProcurementModalityDTO = this.ippModalityNewUpdateForm.value;
+
+        try {
+            let postOrPut;
+            if (ippModality.id != null) { // && ippModality.id != "") {
```

Comment:
not comment in the code!
correct code:
`if (ippModality.id != null && ippModality.id != "") {`